### PR TITLE
Binary_Length.jl

### PR DESCRIPTION
**Pull request by F35H**
The file is short and sweet providing necessary context. I placed it in strings because of how it works, but if there's a more appropriate file please move it or inform me appropriately. Any improvements can be shuttled my way as well.

I've left it out of the file for continuity sake, but I originally developed this as a part of my Youtube Series which you can find on my github.

### CODE REVIEW

**Review by uncomfyhalomacro — COMMENTED** [review]
Should this really be an anonymous function? Otherwise, we cannot add tests plus i think the `println` should be inside the function before maxNum.

**Comment by uncomfyhalomacro** [comment]
@F35H also nice that you have a Youtube series in Julia 😄 i will sub 💯

**Comment by F35H** [comment]
Updated the function. The issue should be resolved. Thanks for reviewing.

**Comment by uncomfyhalomacro** [comment]
@F35H If you have time, please follow [CONTRIBUTING](https://github.com/TheAlgorithms/Julia/blob/main/CONTRIBUTING.md).

File: src/strings/binary_length.jl [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+function binLength()  
+  println("Enter Maximum Number Please")
+  maxNum = parse(UInt64, readline())
+  
+  mod = 0
+  len = 1
+  arr = Array{Int64}(undef, maxNum, 1)
+   
+  Threads.@threads for i in 1:maxNum
```
Comment (uncomfyhalomacro) [inline src/strings/binary_length.jl:16]: I am not quite sure why `Threads.@threads` is needed here. I tested your code with `BencmarkTools` both with unthreaded and threaded.

```julia
julia> @benchmark binary_length("10")
BenchmarkTools.Trial: 10000 samples with 10 evaluations.
 Range (min … max):  1.912 μs … 313.085 μs  ┊ GC (min … max): 0.00% … 98.35%
 Time  (median):     2.117 μs               ┊ GC (median):    0.00%
 Time  (mean ± σ):   2.223 μs ±   3.139 μs  ┊ GC (mean ± σ):  1.39% ±  0.98%

   █▄ ▄▃▂                                                      
  ████████▆▅▃▃▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▁▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂ ▃
  1.91 μs         Histogram: frequency by time         4.5 μs <

 Memory estimate: 736 bytes, allocs estimate: 9.

julia> @benchmark binary_length_unthreaded("10")
BenchmarkTools.Trial: 10000 samples with 986 evaluations.
 Range (min … max):  51.660 ns … 628.397 ns  ┊ GC (min … max): 0.00% … 90.84%
 Time  (median):     54.382 ns               ┊ GC (median):    0.00%
 Time  (mean ± σ):   59.278 ns ±  28.942 ns  ┊ GC (mean ± σ):  2.52% ±  4.93%

  ▅▇█▇▅▄▃▂▁▁▂▃▃▂▂                                              ▂
  ████████████████▆▆▆▅▇▇▇▅▆▅▅▆▆▇██▇███▇█▇▇▆▅▆▅▆▆▆▅▆▆▅▅▅▄▅▄▅▄▄▅ █
  51.7 ns       Histogram: log(frequency) by time       109 ns <

 Memory estimate: 144 bytes, allocs estimate: 1.
 ```
Though this benchmark is not something *we* should actually care about. I just want to know why we use `Threads.@threads` here.

File: src/strings/binary_length.jl [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    if (i == len)
+      mod += 1 
+      len *= 2
+    end
+      
+    arr[i] = mod
+    println(i," ",mod)
+  end
+
```
Comment (uncomfyhalomacro) [inline src/strings/binary_length.jl:25]: add a return statement here so you can add tests for your function

**Review by uncomfyhalomacro — CHANGES_REQUESTED** [review]
Please do make some changes that can actually be tested and also, do follow CONTRIBUTING

**Comment by F35H** [comment]
I've added a second function and more documentation. I've removed the threads macro, renamed variables to be more appropriate, repurposed the functions to make more sense for a general context, and added return statements.

Thank you again. I can be quite the pain to deal with.

**Comment by uncomfyhalomacro** [comment]
@F35H I will merge this later and add the tests myself :) Thanks for the contribution!

**Pull request merged**